### PR TITLE
[RHELC-887] chore: require rebase on push

### PR DESCRIPTION
**Pull request by SpyTec**
This utilizes the hook created by packit to check if the branch is up to
date

Jira Issue: [RHELC-887](https://issues.redhat.com/browse/RHELC-887)

Checklist
- [ ] PR meets acceptance criteria specified in the Jira issue
- [ ] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-]` is part of the PR title 
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending`

### CODE REVIEW

**Comment by SpyTec** [comment]
@TomasTomecek if you can review this to just give us an okay that we can use this

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -32,3 +32,11 @@ repos:
     rev: "1.20.0"
     hooks:
       - id: "tmt-tests-lint"
+
+  - repo: https://github.com/packit/pre-commit-hooks
+    rev: 8db5a24e01b9f54aaa7a800f33c4b9aa619af1b9
```
Comment (r0x0d) [inline .pre-commit-config.yaml:37]: They released 1.0.0 last week: https://github.com/packit/pre-commit-hooks/releases/tag/v1.0.0

```suggestion
    rev: 1.0.0
```

**Review by r0x0d — CHANGES_REQUESTED** [review]
Whenever you have time, @SpyTec, please rebase.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (SpyTec) [reply .pre-commit-config.yaml:37]: Nice! Also @jpopelka pr @TomasTomecek you fine with us using this?

Comment (jpopelka) [reply .pre-commit-config.yaml:37]: sure

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -2,26 +2,14 @@ minimum_pre_commit_version: "2.9.0"
...
-      name: pylint
-      entry: pylint
-      types: [python]
-      language: python
-      additional_dependencies: [pylint==2.15.10]
-      args:
-        [
-          "-sn", # Don't display the score
-          "--rcfile=.pylintrc", # Link to your config file
-        ]
```
Comment (r0x0d) [inline .pre-commit-config.yaml:24]: pylint is gone :(

Comment (SpyTec) [reply .pre-commit-config.yaml:24]: Bad rebase oof

**Review by r0x0d — APPROVED** [review]

**Review by TomasTomecek — COMMENTED** [review]
wow, happy to see you like our pre-commit hook :)

will let folks on my team know you're using it now

(was on a PTO when you pinged me, going through the notification backlog still)

**Pull request merged**